standalone/detectors/fasttrack/src/threadstate.cpp

```diff
@@ -1,26 +1,49 @@
 /*
  * DRace, a dynamic data race detector
  *
- * Copyright 2018 Siemens AG
+ * Copyright 2020 Siemens AG
  *
  * Authors:
  *   Felix Moessbauer <felix.moessbauer@siemens.com>
  *
  * SPDX-License-Identifier: MIT
  */
+
 #include "threadstate.h"
 
 ThreadState::ThreadState(VectorClock::TID own_tid,
                          const std::shared_ptr<ThreadState>& parent)
-    : id(VectorClock::make_id(own_tid)) {
-  vc.insert({own_tid, id});
+    : id(VectorClock::make_epoch(own_tid)), own_tid(own_tid) {
+  vc.insert({make_thread_num(id), id});
   if (parent != nullptr) {
     // if parent exists vector clock
     vc = parent->vc;
   }
 }
 
 void ThreadState::inc_vc() {
-  id++;  // as the lower 32 bits are clock just increase it by ine
-  vc[VectorClock::make_tid(id)] = id;
+  id++;  // as the lower 18 bits are clock just increase it by one
+  vc[make_thread_num(id)] = id;
+}
+
+void ThreadState::set_read_write(std::size_t addr, std::size_t pc) {
+  auto it = read_write.find(addr);
+  if (it == read_write.end()) {
+    // TODO: maybe use std::move avoid copy on pair
+    // should I do sth if traceDepot.GetCurrentElement() is nullptr??
```

Comment:
Maybe discuss this here in the comments or in the PR overview and change/don't change accordingly.